src/main/java/com/microsoft/jenkins/azuread/AzureSecurityRealm.java

```diff
@@ -338,6 +358,7 @@ public boolean canConvert(Class type) {
 
         @Override
         public void marshal(Object source, HierarchicalStreamWriter writer, MarshallingContext context) {
```

Comment:
Hmm why do we need to override marshal? @gavinfish

Reply:
It is a legacy issue. And I do not know much about it either.

Reply:
well if it's a legacy issue than he should not add fromRequest to the legacy code.
There is no chance of unmarshaling/marshaling something with fromRequest.

Reply:
Indeed it will automatically add `<fromrequest>false</fromrequest>` in the config file.

Reply:
Looking into the code base it looks like the ConverterImpl was added to all models...
I am pretty sure nowadays these can be removed without any issues or at least be fixed using the readResolve.
https://wiki.jenkins.io/display/JENKINS/Hint+on+retaining+backward+compatibility